Add working CIS Software Supply Chain controls (#4)
turbot/steampipe-mod-github-compliance: docs/index.md

---
repository: "https://github.com/turbot/steampipe-mod-github-compliance"
---

# GitHub Compliance Mod

> Run security controls across all your AWS accounts to look for resources that are publicly accessible resources, shared with untrusted accounts, have insecure network configurations, and more.

Reviewer: ```suggestion
Run security controls across all your GitHub repositories to look for resources that are publicly accessible resources, shared with untrusted accounts, have insecure network configurations, and more.
```
The description should be updated to use the format of GitHub resources (repositories and organizatons).

<img src="https://raw.githubusercontent.com/turbot/steampipe-mod-github-compliance/staging-branchv0.2/docs/images/github_cis_v100_terminal.png" width="50%" type="thumbnail"/>

## References

[GitHub](https://github.com/) is a provider of Internet hosting for software development and version control using Git.

[Steampipe](https://steampipe.io) is an open source CLI to instantly query cloud APIs using SQL.

[Steampipe Mods](https://steampipe.io/docs/reference/mod-resources#mod) are collections of `named queries`, and codified `controls` that can be used to test current configuration of your cloud resources against a desired configuration.

## Documentation

- **[Benchmarks and controls →](https://hub.steampipe.io/mods/turbot/steampipe-mod-github-compliance/controls)**
- **[Named queries →](https://hub.steampipe.io/mods/turbot/steampipe-mod-github-compliance/queries)**

## Getting started

### Installation

Download and install Steampipe (https://steampipe.io/downloads). Or use Brew:

```sh
brew tap turbot/tap
brew install steampipe
```

> Install the AWS plugin with [Steampipe](https://steampipe.io):

Reviewer: ```suggestion
Install the GitHub plugin with [Steampipe](https://steampipe.io):
```

- Pass in a value on the command line:

  ```shell
  steampipe check control.cis_supply_chain_v100_1_2_3 --var='trusted_repo_admin=["abc22", "bbc22"]'
  ```

- Set an environment variable:

  ```shell
  SP_VAR_trusted_repo_admin='["abc22", "bbc22"]' steampipe check control.cis_supply_chain_v100_1_2_3
  ```

  - Note: When using environment variables, if the variable is defined in `steampipe.spvars` or passed in through the command line, either of those will take precedence over the environment variable value. For more information on variable definition precedence, please see the link below.

These are only some of the ways you can set variables. For a full list, please see [Passing Input Variables](https://steampipe.io/docs/using-steampipe/mod-variables#passing-input-variables).

## Contributing

If you have an idea for additional controls or just want to help maintain and extend this mod ([or others](https://github.com/topics/steampipe-mod)) we would love you to join the community and start contributing.

- **[Join our Slack community →](https://steampipe.io/community/join)** and hang out with other Mod developers.

> Please see the [contribution guidelines](https://github.com/turbot/steampipe/blob/main/CONTRIBUTING.md) and our [code of conduct](https://github.com/turbot/steampipe/blob/main/CODE_OF_CONDUCT.md). All contributions are subject to the [Apache 2.0 open source license](https://github.com/turbot/steampipe-mod-aws-compliance/blob/main/LICENSE).

Reviewer: ```suggestion
Please see the [contribution guidelines](https://github.com/turbot/steampipe/blob/main/CONTRIBUTING.md) and our [code of conduct](https://github.com/turbot/steampipe/blob/main/CODE_OF_CONDUCT.md). All contributions are subject to the [Apache 2.0 open source license](https://github.com/turbot/steampipe-mod-github-compliance/blob/main/LICENSE).
```
